### PR TITLE
[Checkbox#473] Design changes

### PR DESCRIPTION
**Pull request by aycil-alican**
### Pull requests template

This pull request has:
- [ ] Code documentation on publics
- [x] Accessibility
	- [x] Dynamic sizes
	- [x] Identifiers
- [ ] Snapshot Tests
- [x] Unit Tests
- [x] Demo update
- [x] Wiki

### CODE REVIEW

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift [hunk 1]
```diff
@@ -12,17 +12,29 @@ import SwiftUI
 public struct CheckboxGroupView: View {
 
     // MARK: - Private properties
-
+    @available(*, deprecated)
     private var title: String
```
Comment (aycil-alican) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:16]: Maybe we shouldn't mark this as deprecated until we have a form field and we can offer an alternative to the user.

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift [hunk 2]
```diff
@@ -12,17 +12,29 @@ import SwiftUI
 public struct CheckboxGroupView: View {
 
     // MARK: - Private properties
-
+    @available(*, deprecated)
```
Comment (aycil-alican) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:15]: @michael-zimmermann Maybe we shouldn't mark this as deprecated until we have a form field and we can offer an alternative to the user.

Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:15]: @robergro It is replaced by something ? If yes, can you add a message to indicate which new variable we need to use ..?

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift [hunk 3]
```diff
@@ -12,17 +12,29 @@ import SwiftUI
...
+
     private var theme: Theme
+    private var intent: CheckboxIntent
     private var layout: CheckboxGroupLayout
-    private var checkboxPosition: CheckboxPosition
+    private var alignment: CheckboxAlignment
     private var accessibilityIdentifierPrefix: String
-
+    
+    @available(*, deprecated)
```
Comment (aycil-alican) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:27]: @robergro Same here

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift [hunk 4]
```diff
@@ -32,28 +44,63 @@ public struct CheckboxGroupView: View {
     ///   - checkedImage: The tick-checkbox image for checked-state.
     ///   - items: An array containing of multiple `CheckboxGroupItemProtocol`. Each array item is used to render a single checkbox.
     ///   - layout: The layout of the group can be horizontal or vertical.
-    ///   - checkboxPosition: The checkbox is positioned on the leading or trailing edge of the view.
+    ///   - checkboxAlignment: The checkbox is positioned on the leading or trailing edge of the view.
     ///   - theme: The Spark-Theme.
     ///   - accessibilityIdentifierPrefix: All checkbox-views are prefixed by this identifier followed by the `CheckboxGroupItemProtocol`-identifier.
+    @available(*, deprecated)
```
Comment (aycil-alican) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:50]: @robergro Same here, we need to indicate to our consumer which new init they need to use

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift [hunk 5]
```diff
@@ -32,28 +44,63 @@ public struct CheckboxGroupView: View {
...
+    public init(
+        checkedImage: UIImage,
+        items: Binding<[any CheckboxGroupItemProtocol]>,
+        layout: CheckboxGroupLayout = .vertical,
+        alignment: CheckboxAlignment,
+        theme: Theme,
+        intent: CheckboxIntent = .main,
+        accessibilityIdentifierPrefix: String
+    ) {
+        self.title = ""
```
Comment (aycil-alican) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:92]: @robergro You canno't have an optional variable ? It Seems weird to me to have this empty value by default

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift [hunk 6]
```diff
@@ -65,42 +112,106 @@ public struct CheckboxGroupView: View {
...
+    }
+
+    private func makeContentView(maxWidth: CGFloat? = nil) -> some View {
+        var chekboxHeights: [CGFloat] = []
+        return ForEach(self.$items, id: \.id) { item in
             let identifier = "\(self.accessibilityIdentifierPrefix).\(item.id.wrappedValue)"
+            let attributeTextString: String = item.attributedTitle.wrappedValue?.string ?? ""
             CheckboxView(
-                text: item.title.wrappedValue ?? "",
+                text: item.title.wrappedValue ?? attributeTextString,
```
Comment (aycil-alican) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:177]: @robergro Same here, why you have this default property if the string is nil ? The text should be optional here I think

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 5 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:92]: you are right, I missed to refactor here. It was like that. I changed title as an optinal and removed   default value

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 6 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:177]: In swiftui component we dont suport attribute text for now and both of swiftui and uikit component use same protocol to initiliaze checkbox group. title and attributedTitle were optinal on CheckboxGroupItemProtocol, and while creating swiftui single checkbox  item at least there must be string value that is why first I checked attribute text if people use it wrongly in swiftui then I assigned default value.

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 2 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:15]: we will create form field to add title and supplementary texts for components. designers said that consumer should manage title and supplementary texts themself. It isnt part of checkboxgroup component. That is why I deprecated this. In last case I dont know what I should do. I can keep this as It is or I can add comment to deprecated reason

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 3 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:27]: I will update it according to the comment above.

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 4 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:50]: I will update it according to the comment above.

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 5 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:92]: fixed

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 2 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:15]: @michael-zimmermann @robergro ?

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 6 above)
Comment (robergro) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:177]: Oh no 😣
We not suppose have a NSAttributedString for SwiftUI... This is why Michael create a Either...
I don't know how we can update that...

For the other component, you can check the switch for example, we have the NSattributedString for UIKit and AttributedString with SwiftUI and on the ViewModel we have an AttributedStringEither

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 2 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:15]: I talked with @LouisBorleeAdevinta and I will remove deprecated line here until we can provide another solution with consumer.

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift (hunk 5 above)
Comment (aycil-alican) [reply core/Sources/Components/Checkbox/View/SwiftUI/CheckboxGroupView.swift:92]: I pushed it I dont know why you don see but here is my last commit; https://github.com/adevinta/spark-ios/pull/603/commits/9271659473e281ab5286a7d9b830c0bf2e82f49c

File: core/Sources/Components/Checkbox/Enum/SelectButtonState.swift [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+//  CheckboxState.swift
+//  SparkCore
+//
+//  Created by janniklas.freundt.ext on 11.04.23.
+//  Copyright © 2023 Adevinta. All rights reserved.
+//
+
+import Foundation
+
+@available(*, deprecated, message: "isEnabled: Bool parameter will be used instead od this.")
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Checkbox/Enum/SelectButtonState.swift:11]: typo on "of"

File: core/Sources/Components/Checkbox/UseCase/CheckboxGetSpacingUseCase.swift [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+struct CheckboxGetSpacingUseCase: CheckboxGetSpacingUseCaseable {
+    func execute(layoutSpacing: LayoutSpacing, alignment: CheckboxAlignment) -> CGFloat {
+        switch alignment {
+        case .left: return layoutSpacing.medium
+        case .right: return layoutSpacing.xxxLarge
+        }
+    }
+}
+
+//switch self.viewModel.alignment {
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Checkbox/UseCase/CheckboxGetSpacingUseCase.swift:24]: Remove comments

File: core/Sources/Components/Checkbox/View/SwiftUI/CheckboxView.swift [hunk 9]
```diff
@@ -44,80 +41,58 @@ public struct CheckboxView: View {
 
     @ObservedObject var viewModel: CheckboxViewModel
 
-    var colors: CheckboxColorables {
-        return self.viewModel.colors
-    }
-
     // MARK: - Private Properties
+    
+//    private var spacing: LayoutSpacing {
```
Comment (LouisBorleeAdevinta) [inline core/Sources/Components/Checkbox/View/SwiftUI/CheckboxView.swift:46]: remove

**Review by LouisBorleeAdevinta — APPROVED** [review]

**Pull request merged**